UnitTests/UnitTests.UWP/Extensions/Test_FrameworkElementExtensions.cs

```diff
@@ -0,0 +1,95 @@
+                    xmlns=""http://schemas.microsoft.com/winfx/2006/xaml/presentation""
+                    xmlns:x=""http://schemas.microsoft.com/winfx/2006/xaml""
+                    xmlns:controls=""using:Microsoft.Toolkit.Uwp.UI.Controls""
+                    xmlns:ui=""using:Microsoft.Toolkit.Uwp.UI"">
+
+                    <Grid x:Name=""OuterGrid"" Width=""200"" Background=""Khaki"">
+                        <Button x:Name=""InnerButton"" ui:FrameworkElementExtensions.AncestorType=""Grid"" />
+                    </Grid>
+
+                </Page>") as Page;
+
+                Assert.IsNotNull(treeRoot, "Could not load XAML tree.");
+                await SetTestContentAsync(treeRoot);
+
+                // Need to simulate loading the control (and not just rely upon XamlReader.Load)
+                var frame = new Frame();
+                frame.Navigate(treeRoot.GetType());
+                Assert.IsInstanceOfType(frame.Content, typeof(Page));
+
+                var btn = treeRoot.FindChild("InnerButton") as Button;
+
+                Assert.IsNull(btn.Parent);
+
+                btn.SetValue(FrameworkElementExtensions.AncestorProperty, new object());
+
+                Assert.IsNull(btn.Parent);
+            });
+        }
+
+        [TestCategory("FrameworkElementExtensions")]
+        [TestMethod]
+        public async Task Test_Ancestor_WeakRef_UnloadGrid()
+        {
+            // Need to sim loading the control, and not just load the XAML via XamlReader
+            await App.DispatcherQueue.EnqueueAsync(async () =>
+            {
+                var treeRoot = XamlReader.Load(
+           @"<Page
+                    xmlns=""http://schemas.microsoft.com/winfx/2006/xaml/presentation""
+                    xmlns:x=""http://schemas.microsoft.com/winfx/2006/xaml""
+                    xmlns:controls=""using:Microsoft.Toolkit.Uwp.UI.Controls""
+                    xmlns:ui=""using:Microsoft.Toolkit.Uwp.UI"">
+
+                    <Grid x:Name=""OuterGrid"" Width=""200"" Background=""Khaki"">
+                        <Button x:Name=""InnerButton"" ui:FrameworkElementExtensions.AncestorType=""Grid"" />
+                    </Grid>
+
+                </Page>") as Page;
+
+                Assert.IsNotNull(treeRoot, "Could not load XAML tree.");
+                await SetTestContentAsync(treeRoot);
+
+                // Need to simulate loading the control (and not just rely upon XamlReader.Load)
+                var frame = new Frame();
+                frame.Navigate(treeRoot.GetType());
+                Assert.IsInstanceOfType(frame.Content, typeof(Page));
+
+                var grid = treeRoot.FindChild("OuterGrid") as Grid;
+                var button = treeRoot.FindChild("InnerButton") as Button;
+
+                button.SetValue(FrameworkElementExtensions.AncestorProperty, new object());
+
+                treeRoot.Unloaded += (sender, e) =>
+                {
+                    Assert.AreEqual(grid, null);
```

Comment:
Wonder if this is getting called late and causing a problem as immediately after this the test ends...

We're also waiting for the Unloaded event from the content itself as part of the VisualUITestBase Cleanup method, so we may need to try forcing your content to unload first before letting the test end?

Maybe try calling `await SetTestContentAsync(null)` after this? The Clean-up method also shows using a `TaskCompletionSource` too which we could use to await the event to ensure we got the unloaded callback and the assert is checked.